Datastore/src/Operation.php

```diff
@@ -497,6 +503,15 @@ public function runQuery(QueryInterface $query, array $options = [])
             if (isset($res['query'])) {
                 $runQueryObj = new Query($this->entityMapper, $res['query']);
             }
+            if (isset($res['query']['limit'])) {
+                $remainingLimit = $res['query']['limit'];
+            }
+            if (isset($remainingLimit['value'])) {
```

Comment:
This looks redundant - `['value']` could be coming from `$res['query']['limit']`, or from `$query->queryObject()['limit']`. So why set it above just to set it again here? It seems to me like what you'd want to do is set `$remainingLimit` to `null`, above, and then simply make this:

```php
            if (isset($res['query']['limit']['value'])) {
                $remainingLimit = $res['query']['limit']['value'];
            }
```

But maybe there's something I'm missing.

Reply:
There are several cases here:
1. In a standard Query, (either GRPC or REST) we don't get a `query` key in the response. So, we set `$remainingLimit` using `$query->queryObject()['limit']`: https://github.com/googleapis/google-cloud-php/blob/71ac41c2a0f036a143e4467984955d1265c56958/Datastore/src/Operation.php#L472
2. In GqlQuery, we don't have limit from the query. So while using REST, we get limit in `$res['query']['limit']`: https://github.com/googleapis/google-cloud-php/blob/71ac41c2a0f036a143e4467984955d1265c56958/Datastore/src/Operation.php#L507
3. In GqlQuery while using GRPC, we get limit in `$res['query']['limit']['value']`: https://github.com/googleapis/google-cloud-php/blob/71ac41c2a0f036a143e4467984955d1265c56958/Datastore/src/Operation.php#L509-L510